httpcore/_async/http_proxy.py

```diff
@@ -144,8 +161,10 @@ async def _tunnel_request(
             # [proxy-headers]
             target = b"%b:%d" % (url[1], url[2])
             connect_url = self.proxy_origin + (target,)
+            connect_headers = [(b"Host", target), (b"Accept", b"*/*")]
+            connect_headers = merge_headers(connect_headers, self.proxy_headers)
             proxy_response = await proxy_connection.request(
-                b"CONNECT", connect_url, headers=self.proxy_headers, timeout=timeout
+                b"CONNECT", connect_url, headers=connect_headers, timeout=timeout
```

Comment:
We should probably remove the [setting of `host` headers in the tests](https://github.com/encode/httpcore/blob/master/tests/async_tests/test_interfaces.py#L188-L193) to cover things are working as expected.

Reply:
Good call!